### PR TITLE
CI Update

### PR DESCRIPTION
**Pull request by ColdenCullen**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-Checks:          'clang-diagnostic-*,clang-analyzer-*,readability-*,modernize-*,bugprone-*,misc-*,google-runtime-int,fuchsia-restrict-system-includes,-clang-analyzer-valist.Uninitialized,-clang-analyzer-security.insecureAPI.rand,-clang-analyzer-alpha.*,-readability-else-after-return,-misc-unused-parameters'
+Checks:          'clang-diagnostic-*,clang-analyzer-*,readability-*,modernize-*,bugprone-*,misc-*,google-runtime-int,fuchsia-restrict-system-includes,-clang-analyzer-valist.Uninitialized,-clang-analyzer-security.insecureAPI.rand,-clang-analyzer-alpha.*,-readability-else-after-return,-readability-magic-numbers,-misc-unused-parameters'
```
Comment (justinboswell) [inline .clang-tidy]: Have we not standardized this? Should we? As in like, here's a script:
```sh
curl <url to common .clang-tidy>
clang-tidy
```

File: source/hpack.c [hunk 2]
```diff
@@ -502,26 +502,29 @@ int aws_hpack_resize_dynamic_table(struct aws_hpack_context *context, size_t new
...
-        above_block_size * sizeof(struct aws_http_header));
-
-    /* Copy as much of below block as possible */
-    const size_t free_blocks_available = new_max_elements - above_block_size;
-    const size_t old_blocks_to_copy = context->dynamic_table.max_elements - above_block_size;
-    const size_t below_block_size =
-        free_blocks_available > old_blocks_to_copy ? old_blocks_to_copy : free_blocks_available;
-    if (below_block_size) {
+    /* Don't bother copying data if old buffer was of size 0 */
+    if (AWS_UNLIKELY(context->dynamic_table.max_elements > 0)) {
```
Comment (justinboswell) [inline source/hpack.c]: Is this unlikely, or likely? Is the dynamic table THAT unlikely?

File: .clang-tidy (hunk 1 above)
Comment (ColdenCullen) [reply .clang-tidy]: Good idea! I'm gonna copy paste this script for later.

Joke's on you though, a lot of these images don't have curl :P

File: source/hpack.c (hunk 2 above)
Comment (ColdenCullen) [reply source/hpack.c]: Oh fuck I got that backwards

File: .clang-tidy (hunk 1 above)
Comment (justinboswell) [reply .clang-tidy]: We could fix that...we have the technology!

**Pull request merged**